compositor_pipeline/src/queue/audio_queue.rs

```diff
@@ -153,6 +153,9 @@ impl AudioQueueInput {
         pts_range: (Duration, Duration),
         queue_start: Instant,
     ) -> PipelineEvent<Vec<InputSamples>> {
+        // ignore result, we only need to ensure samples are enqueued
+        self.check_ready_for_pts(pts_range, queue_start);
```

Comment:
It's awful. I hate it. Over and over we rely on some internal behavior of this function (enqueuing frames / reading them from the channel).

I'll try try to simplify this logic, cause reading all the queue code slowly feels like reading brainfuck.

Reply:
I would prefer having a function like `get_frames` and calling it here and in the `check_ready_for_pts` function

Reply:
that would not work with current architecture:
- get frames needs to return function whether it's ready or not
- currently get frames relies that returned frame will be used (it's needed for example to check whether EOS was sent)

And if you are suggesting to change the above, then your suggestion boils down to almost full rewrite of queuing code.

We can talk about it more at the office, topic is to complex for discussion via comments.